Lisää poikkeusten yhteenveto kojelautaan

jarnovayrynen opened the pull request:
Lisää poikkeusten yhteenveto kojelautaan
Tyylittele yhteenvetorivit kauniimmin

jarnovayrynen committed:
Lisää poikkeusten yhteenveto kojelautaan

rckopteri commented on src/cljs/harja/tiedot/urakkatilanne/kojelauta.cljs:
```diff
@@ -31,6 +31,27 @@
+(defn poikkeusten-yhteenveto
+  [urakat]
+  (let [kaikkien-urakoiden-lkm (count urakat)
+        urakat-joissa-avoimia-laatupoikkeamia (count (filter (fn [rivi]
```
Ei merkittävää, mutta tuossa countilla otetaan vain urakoiden lukumäärä, ei itse urakkaa, vaikka parametri niin viittaakin.

jarnovayrynen replied:
Joo, näin juuri. Koetin tuossa pitää lyhyenä sitä sanamuotoa.. Mietinpä vielä muutanko.

rckopteri commented on src/cljs/harja/tiedot/urakkatilanne/kojelauta.cljs:
```diff
@@ -39,10 +60,10 @@
-                                        "Lupaukset kirjattu:" (str urakat-joissa-pisteet-syotetty " (" (fmt/prosentti-opt (* 100 (/ urakat-joissa-pisteet-syotetty kaikkien-urakoiden-lkm))) ")")]])]
+                                      [:span.lupaustiedot
+                                       [yleiset/tietoja {:class "body-text kojelauta-tietoja"}
+                                        "Lupaukset puuttuvat" (str urakat-joista-tieto-puuttuu " (" (fmt/prosentti-opt (* 100 (/ urakat-joista-tieto-puuttuu kaikkien-urakoiden-lkm)) 0) ")")
```
Tämän (fmt/prosentti-opt (* 100 (/ urakat-joista-tieto-puuttuu kaikkien-urakoiden-lkm)) 0) voisi melkein käärästä erilliseksi funktioksi. Tyyliin
(defn- esita-prosenttina 
"Esitetään prosenttina kahdesta luvusta tehty jakolasku. Ensimmäinen parametri on osamäärä ja toiseen parametriin annetaaan kokonaismäärä."
[osamaara kokonaismaara]
(fmt/prosentti-opt (* 100 (/ osamaara kokonaismaara)) 0))

jarnovayrynen replied:
Hyvä pointti. Löysin lopulta Harjan kätköistä tällaisen mitä nyt käytän:
(math/osuus-prosentteina urakat-joissa-ks-aloittamatta kaikkien-urakoiden-lkm)
Hyvä ettei tule kylvettyä turhaa matematiikkaa joka paikkaan. Kiitos hyvästä kommentista.

jarnovayrynen committed:
Poista tierekisteri-kansio sisältöineen

Tierekisteri on poistunut käytöstä, aika poistaa Harjan reposta viittaukset näihin. Velho on uudempi järjestelmä, joka kantaa entisen Tierekisterin vastuita.

jarnovayrynen committed:
Käytä math/osuus-prosentteina funktiota osuuksien laskentaan

jarnovayrynen committed:
Tue näppisnavigointia kojelaudan linkeissä

jarnovayrynen committed:
Muuta termi Tierekisteriosoite -> Tieosoite

jarnovayrynen committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into dashboard_poikkeusten_yhteenveto
